adapters/rest-vertx/src/main/java/org/eclipse/hono/adapter/rest/VertxBasedRestProtocolAdapter.java

```diff
@@ -48,7 +49,16 @@ protected final void addRoutes(final Router router) {
     }
 
     private void setupBasicAuth(final Router router) {
-        router.route().handler(BasicAuthHandler.create(getCredentialsAuthProvider(), getConfig().getRealm()));
+        router.route().handler(new HonoAuthHandlerImpl(getCredentialsAuthProvider(), getConfig().getRealm()) {
+            @Override
+            protected void processException(RoutingContext ctx, Throwable exception) {
```

Comment:
in the original `AuthHandlerImpl` class this method is declared *private* so we won't be able to override it like this.
Shouldn't we instead try to register an exception handler which recognizes the `ServiceInvocationException`?

Reply:
I requested a change to protected in Vert.x (in the ticket) - I think this would be nice for others also, to handel different types of exceptions. I would suggest to implement your solution only, if the Vert.x team does not want to make this change.